providers/flagd/pkg/provider.go

```diff
@@ -118,6 +139,79 @@ func WithHost(host string) ProviderOption {
 	}
 }
 
+// WithoutCache disables caching
+func WithoutCache() ProviderOption {
+	return func(p *Provider) {
+		p.cacheEnabled = false
+		p.booleanCache = nil
+		p.stringCache = nil
+		p.floatCache = nil
+		p.interfaceCache = nil
+		p.intCache = nil
+	}
+}
+
+// WithBasicInMemoryCache applies a basic in memory cache store (with no memory limits)
+func WithBasicInMemoryCache() ProviderOption {
+	return func(p *Provider) {
+		p.booleanCache = cache.NewInMemory[string, of.BoolResolutionDetail]()
+		p.intCache = cache.NewInMemory[string, of.IntResolutionDetail]()
+		p.stringCache = cache.NewInMemory[string, of.StringResolutionDetail]()
+		p.floatCache = cache.NewInMemory[string, of.FloatResolutionDetail]()
+		p.interfaceCache = cache.NewInMemory[string, of.InterfaceResolutionDetail]()
+
+		p.cacheEnabled = true
+	}
+}
+
+// WithLRUCache applies least recently used caching (github.com/hashicorp/golang-lru).
+// The provided size is the limit of the number of cached values for each type of flag. Once the limit is reached each
```

Comment:
I really thought "size" was "size in bytes", but seems it's actually "number of entries". The LRU cache doc isn't very clear about this.